### PR TITLE
Final c 1.0

### PR DESCRIPTION
**Pull request by jeromekelleher**
Final changes for tskit API 1.0.0

### CODE REVIEW

**Comment by molpopgen** [comment]
Is there a test for copying table collections?  I believe the copies are failing, which I discovered while working on #98.  I am prepping a PR with a test to check.

**Comment by jeromekelleher** [comment]
I'll have a look at the test @molpopgen, thanks.

In the mean time, I've made a fair few changes here. Most are uncontroversial improvements I think, but I'm not  certain about changing ``tsk_table_collection_position_t`` to ``tsk_bookmark_t``. Seems much more intuitive to me, but I'm happy to be disagreed with! @molpopgen, @petrelharp, any thoughts?

**Comment by jeromekelleher** [comment]
This should be the last of the code changes we need on the tables API, so I'm going to merge. Rather than tagging 1.0.0 right now I think we can tag 0.99.2, giving a bit of room for changes  (there's still a lot to be nailed down in terms of the tree and tree sequence APIs, which I don't have time to do now). I'll follow up with a PR to make some final documentation updates on how to use tskit.

**Comment by petrelharp** [comment]
:celebration: :fireworks:

File: c/CHANGELOG.rst [hunk 1]
```diff
@@ -11,6 +11,14 @@ Refinements to the C API as we move towards 1.0.0. Changes:
 - Add ``tsk_flags_t`` typedef and consistently use this as the type used to
   encode bitwise flags. To avoid confusion, functions now have an ``options``
   parameter.
+- Rename ``tsk_table_collection_position_t`` to ``tsk_bookmark_t``.
```
Comment (petrelharp) [inline c/CHANGELOG.rst]: :thumbsup:

**Pull request merged**